index.js

```diff
@@ -0,0 +1,99 @@
+function take(arr, n) {
+  var newArr = [];
+  var amount = n;
+  if (n > arr.length) {
+    amount = arr.length;
+  }
+  for (var i = 0; i < amount; i++) {
+    newArr.push(arr[i]);
+  }
+  return newArr;
+}
+
+
+function skip(arr, n) {
+  var newArr = [];
+  if (n > arr.length) {
+    return newArr;
+  }
+  for (var i = n; i < arr.length; i++) {
+    newArr.push(arr[i]);
+  }
+  return newArr;
+}
+
+function map(arr, callback) {
+  var newArr = [];
+  for (var i = 0; i < arr.length; i++) {
+    newArr.push(callback(arr[i], i, arr));
+  }
+  return newArr;
+}
+
+function reduce(arr, callback, initialValue) {
+  var initial = initialValue;
+  var i = 0;
+  if (!initial) {
```

Comment:
What if `initialValue = 0`, then this check will `true`
I think it will be better to check if `initial` is `undefined` here